File: WebClient SDK/src/default/cards/removeTagsCardStoreExtension.ts
import { CardStoreExtension, ICardStoreExtensionContext } from 'tessa/cards/extensions';
import { extension } from '@tessa/application';

@extension()
export class RemoveTagsCardStoreExtension extends CardStoreExtension {
  async beforeRequest(context: ICardStoreExtensionContext): Promise<void> {
    if (!context.validationResult.isSuccessful) {
      return;
    }

    const info = context.request.tryGetCard()?.tryGetInfo();
    if (!info) {
      return;
    }

    delete info['TagsForCard'];
  }
}

File: Extensions/Tessa.Extensions.Default.Client/Cards/RemoveTagsCardStoreExtension.cs
using System.Threading.Tasks;
using Tessa.Cards.Extensions;
using Tessa.Tags;

namespace Tessa.Extensions.Default.Client.Cards
{
    public sealed class RemoveTagsCardStoreExtension
        : CardStoreExtension
    {
        public override Task BeforeRequest(ICardStoreExtensionContext context)
        {
            if (!context.ValidationResult.IsSuccessful() ||
                context.Request.TryGetCard()?.TryGetInfo() is not { } info)
            {
                return Task.CompletedTask;
            }

            TagsForCard.Remove(info);

            return Task.CompletedTask;
        }
    }
}
